widget/diagramwidget/geometry/r2/line.go

```diff
@@ -0,0 +1,128 @@
+package r2
+
+// Line describes a line in R2
+//
+//	(1) A.X,A.Y  +
+//	              \
+//	               \
+//	                \
+//	                 \
+//	                  + A.X+S.X,A.Y+S.Y (2)
+type Line struct {
+	// A defines the basis point of the line
```

Comment:
Like box you can use our provided types

Reply:
See above comment

Reply:
Again, the purpose of the r2 package is to perform operations on geometric shapes. Short of revising the fyne geometry package (which is beyond the scope of the DiagramWidget), I don't see another alternative. If the fyne geometry package is updated (a separate project), I would be more than happy to update the DiagramWidget to refer to it rather than the r2 package.